plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -1697,7 +1696,7 @@ private static void registerJDKInternalIntrinsics(final CompilationContext ctxt)
 
             ValueType expectType = newValue.getType();
             Value result = builder.cmpAndSwap(builder.instanceFieldOf(builder.referenceHandle(input), field), lf.zeroInitializerLiteralOfType(expectType),
-                newValue, MemoryAtomicityMode.VOLATILE, MemoryAtomicityMode.MONOTONIC, CmpAndSwap.Strength.STRONG);
+                newValue, GlobalAcquire, SingleOpaque, CmpAndSwap.Strength.STRONG);
```

Comment:
I'm not certain the original definition here is strictly correct, but to be equivalent to the original code, this would have to be `GlobalSeqCst`/`SingleOpaque`.